src/main/java/io/apicurio/datamodels/core/validation/rules/other/SecurityRequirementUniquenessValidationRule.java

```diff
@@ -0,0 +1,157 @@
+        return currentSecurityRequirementScopes.equals(securityRequirementToCompareScopes);
+    }
+
+    private List<String> currentSecurityRequirementScopes = new ArrayList<String>();
+    private List<String> securityRequirementToCompareScopes = new ArrayList<String>();
+
+    /**
+     * @param currentSecurityRequirement
+     * @param securityRequirementToCompare
+     */
+    private void getScopes(SecurityRequirement currentSecurityRequirement,
+            SecurityRequirement securityRequirementToCompare) {
+        System.out.println("Getting scopes...");
+        //get the scopes for each of the two security requirements that I'm comparing
+        currentSecurityRequirement.getSecurityRequirementNames()
+                .forEach(securityRequirementName -> currentSecurityRequirementScopes
+                        .addAll(currentSecurityRequirement.getScopes(securityRequirementName)));
+
+        securityRequirementToCompare.getSecurityRequirementNames()
+                .forEach(securityRequirementName -> securityRequirementToCompareScopes
+                        .addAll(securityRequirementToCompare.getScopes(securityRequirementName)));
+
+        System.out.println("Finished getting scopes...");
+    }
+
+    /**
+     * @param currentSecurityRequirement
+     * @param allRequirements
+     */
+    private void checkSecurityRequirementsForNode(SecurityRequirement currentSecurityRequirement,
+            List<SecurityRequirement> allRequirements) {
+        ArrayList<SecurityRequirement> clonedList = new ArrayList<SecurityRequirement>();
+        clonedList.addAll(allRequirements);
+        clonedList.remove(currentSecurityRequirement); // remove the current one to examine siblings
+
+        for (int j = 0; j < clonedList.size(); j++) {
+            // get the SecurityRequirement to compare against and also get each requirement name
+            SecurityRequirement securityRequirementToCompare = clonedList.get(j);
+            List<String> securityRequirementNames = currentSecurityRequirement.getSecurityRequirementNames();
+            List<String> securityRequirementToCompareNames = securityRequirementToCompare
+                    .getSecurityRequirementNames();
+
+            System.out.println("\nchecking for comparisons of: " + securityRequirementNames);
+            System.out.println("checking against: " + securityRequirementToCompareNames);
+
+            if (checkForDuplicateSecurityRequirement(currentSecurityRequirement,
+                    securityRequirementToCompare)) { // initial check for duplicate security requirement names
+                System.out.println("Duplicate Security Requirement Found");
+                if (securityRequirementNames.size() == 1 && securityRequirementNames.get(0).equals("api_key")
+                        || (securityRequirementToCompareNames.size() == 1
+                                && securityRequirementToCompareNames.get(0).equals("api_key"))) {
```

Comment:
I confess I don't understand this at all.  The `api_key` name is not significant in the context of security requirements.  The name just references a declared Security Scheme, which itself can be arbitrarily named.  So I can't think of any reason why you'd have a hard-coded name of a security requirement in this code.

It seems like maybe you were trying to break out in cases where the security requirement is of a type (e.g. apiKey or basic auth) where scopes are not needed.  But that's not what you want to do in this validator.  Another validator is responsible for reporting a problem if someone uses scopes on a security type that doesn't support it.

Reply:
ok, I think it was more about just break out and don't waist the time to bother check if the scopes match as they will/should be empty in this case. But If thats for another validator to worry about then I will remove this. (along with the fact that I forgot about basic auth anyway)

Reply:
Sounds good, but just to be clear, you weren't actually checking for apiKey auth either.  You were checking the **name** of the auth, which is something provided by the person who created the OpenAPI doc.  That user could have named it "giraffe" or "skyscraper".  Or worse, the user could have named it "api_key" but then set the type of auth to be OAuth.  In other words, the name of the auth inside the security requirement does not tell you what kind of auth it is.

Reply:
The only way to do that, would be to use the name to look up the SecurityScheme node in the document and check its `type` property.

Reply:
ok np, I understand it now better